python/packages/core/agent_framework/azure/_search_provider.py

```diff
@@ -0,0 +1,538 @@
+                model_name=self.model_name,  # Underlying model name (e.g., "gpt-4o")
+                api_key=self.azure_openai_api_key,  # Optional: for API key auth instead of managed identity
+            )
+
+            # Note: SDK uses KnowledgeAgent class name, but this represents a Knowledge Base
+            knowledge_base = KnowledgeAgent(
+                name=knowledge_base_name,
+                description=f"Knowledge Base for multi-hop retrieval across {self.index_name}",
+                models=[KnowledgeAgentAzureOpenAIModel(azure_open_ai_parameters=aoai_params)],
+                knowledge_sources=[
+                    KnowledgeSourceReference(
+                        name=knowledge_source_name,
+                        include_references=True,
+                        include_reference_source_data=True,
+                    )
+                ],
+                output_configuration=KnowledgeAgentOutputConfiguration(
+                    modality=KnowledgeAgentOutputConfigurationModality.ANSWER_SYNTHESIS,
+                    attempt_fast_path=True,
+                ),
+                request_limits=KnowledgeAgentRequestLimits(
+                    max_output_size=10000,
+                    max_runtime_in_seconds=60,
+                ),
+                retrieval_instructions=self.retrieval_instructions,
+            )
+            await self._index_client.create_agent(knowledge_base)
+
+        self._knowledge_base_initialized = True
+
+    async def _agentic_search(self, query: str) -> str:
+        """Perform agentic retrieval with multi-hop reasoning using Knowledge Bases.
+
+        NOTE: This mode is significantly slower than semantic search and should
+        only be used for complex queries requiring cross-document reasoning.
+
+        This method uses Azure AI Search Knowledge Bases which:
+        1. Analyze the query and plan sub-queries
+        2. Retrieve relevant documents across multiple sources
+        3. Perform multi-hop reasoning with an LLM
+        4. Synthesize a comprehensive answer with references
+
+        Args:
+            query: Search query text.
+
+        Returns:
+            Synthesized answer from the Knowledge Base.
+        """
+        # Ensure Knowledge Base is initialized
+        await self._ensure_knowledge_base()
+
+        # Type narrowing: knowledge_base_name is validated in __init__ for agentic mode
+        # Using cast() for type checker - actual validation happens in __init__
+        knowledge_base_name = cast(str, self.knowledge_base_name)
+
+        # Create retrieval request with query as a conversation message
+        # Note: SDK uses KnowledgeAgent class names, but represents Knowledge Base operations
+        retrieval_request = KnowledgeAgentRetrievalRequest(
+            messages=[
+                KnowledgeAgentMessage(
+                    role="user",
+                    content=[KnowledgeAgentMessageTextContent(text=query)],
```

Comment:
If you're given a list of messages above, why not pass the list of messages here instead of flattening into a single query?